tests/clickhouse-test

```diff
@@ -36,6 +37,12 @@ MESSAGES_TO_RETRY = [
     "ConnectionPoolWithFailover: Connection failed at try",
 ]
 
+def stop_tests():
```

Comment:
I tried to do this earlier with a separate process group + atexit callbacks:
https://github.com/ClickHouse/ClickHouse/commit/573983d4079b280149699c5fe2b522f4ae4c9fb4#diff-3a359de18cacf146f406a7ae332fb47196aa5e0aa430eb4b157a202a3cb8e6e3R578

But that commit was later reverted because it also tried to switch to multithreading instead of multiprocessing, and that didn't go good. `SIG_IGN` and `SIG_DFL` were broken then (https://bugs.python.org/issue23395), now they are fixed but not quite but maybe it's not relevant for us.

Reply:
>I tried to do this earlier with a separate process group + atexit callbacks:
573983d#diff-3a359de18cacf146f406a7ae332fb47196aa5e0aa430eb4b157a202a3cb8e6e3R578

Thanks for the reference!

> SIG_IGN and SIG_DFL were broken then (https://bugs.python.org/issue23395), now they are fixed but not quite but maybe it's not relevant for us.

I didn't see any issues while testing locally, and I think that it is better to kill tests and get some (even non-standard) error instead of leaving tests in background, since hung check will find them and report, thoughts?